src/gpuarray_buffer_opencl.c

```diff
@@ -113,6 +172,27 @@ cl_ctx *cl_make_ctx(cl_context ctx, int flags) {
     return NULL;
   }
   res->refcnt--; /* Prevent ref loop */
+
+  /* Create per-context OpenCL preamble */
+
+  // Create a dummy kernel and check GA_KERNEL_PROP_PREFLSIZE
+  rlk[0] = dummy_kern;
+  len = sizeof(dummy_kern);
+  // this dummy kernel does not require a CLUDA preamble
+  m = cl_newkernel((gpucontext *)res, 1, rlk, &len, "kdummy", 0, NULL, 0, &ret, NULL);
+  assert((m != NULL) && "Cannot create a dummy OpenCL kernel!");
+  ret = cl_property((gpucontext *)res, NULL, m, GA_KERNEL_PROP_PREFLSIZE, &warp_size);
+  if (ret != GA_NO_ERROR) warp_size = 128;
+
+  // Write the preferred workgroup multiple as GA_WARP_SIZE in preamble
+  len = snprintf(NULL, 0, CL_CONTEXT_PREAMBLE, warp_size);
```

Comment:
This will not work on windows.  Please use strb_appendf() which handles this correctly on all platforms.